engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -825,10 +826,14 @@ case class WorkflowActor(workflow: WorkflowDescriptor)
       case _ => false
     }
 
+    lazy val hasAnythingFailed = executionStore.values exists { x => x == Failed || x == Aborted }
+
+    lazy val workflowFailFastTrigger = workflowFailureMode == NoNewCalls && hasAnythingFailed
```

Comment:
Hmm I don't see why ? I can't see how we would have different classes that implement this behaviour differently in Cromwell ?
there could be a 
`def shouldFail(hasAnythingFailed: Boolean)` method in `WorkflowFailureMode` that would return `hasAnythingFailed` for `NoNewCalls` and `false` for `ContinueWhilePossible`